homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -6,58 +6,105 @@
 import voluptuous as vol
 
 from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
     CONF_ADDRESS,
     CONF_DEVICE,
     CONF_NAME,
     DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
     POWER_WATT,
+    TEMP_CELSIUS,
 )
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.entity import Entity
+
+from .aurora_device import AuroraDevice
+from .const import ATTR_SERIAL_NUMBER, DEFAULT_ADDRESS, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_ADDRESS = 2
-DEFAULT_NAME = "Solar PV"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_DEVICE): cv.string,
         vol.Optional(CONF_ADDRESS, default=DEFAULT_ADDRESS): cv.positive_int,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NAME, default="Solar PV"): cv.string,
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Aurora ABB PowerOne device."""
+def setup_platform(hass, config: ConfigEntry, add_entities, discovery_info=None):
+    """Set up single sensor based on configuration.yaml (DEPRECATED)."""
     devices = []
     comport = config[CONF_DEVICE]
-    address = config[CONF_ADDRESS]
-    name = config[CONF_NAME]
-
+    address = config.get(CONF_ADDRESS, DEFAULT_ADDRESS)
+    name = config.get(CONF_NAME, "Solar PV")
+    _LOGGER.warning(
+        "DEPRECATED: setting up %s via configuration.yaml will "
+        "soon be unsupported.  Please remove the entries from your "
+        "configuration.yaml file and the set up the integration via the UI",
+        DOMAIN,
+    )
     _LOGGER.debug("Intitialising com port=%s address=%s", comport, address)
     client = AuroraSerialClient(address, comport, parity="N", timeout=1)
 
-    devices.append(AuroraABBSolarPVMonitorSensor(client, name, "Power"))
+    config.data = {"device_name": name}
```

Comment:
You can't just set properties on a dict. You have not tested this code.

Reply:
It seems to work now, maybe due to the other changes.  Please advise if this is still not ok.

Reply:
This is still a problem

Reply:
I have fixed this now and added tests. Thanks for spotting this!